File: 프로그래머스/1/12931. 자릿수 더하기/자릿수 더하기.cs
using System;

public class Solution {
    public int solution(int n) {
        int answer = 0;
        int number = n;
        int temp = 0;
        int i =1;
        while(number / i != 0)
        {
            temp = number/i;
            answer += temp % 10;
            i = i * 10;
        }
        return answer;
    }
}

File: 프로그래머스/1/12931. 자릿수 더하기/자릿수 더하기.java
import java.util.*;

public class Solution {
    public int solution(int n) {
        int answer = 0;
        int line = 1;
        
        while(n / line != 0) {
            answer += (n / line) % 10;
            System.out.print(n % line + " ");
            line *= 10;
        }
        

        return answer;
    }
}